stk/molecular/molecules/macrocycle.py

```diff
@@ -35,70 +34,60 @@ def cycle_atoms(self, conformer=-1):
-        :class:`.list` of :class:`.int`
+        :class:`list` of :class:`int`
             Atom ids of the atoms comprising the largest ring.
 
         """
 
         ssr = rdkit.GetSymmSSSR(self.mol)
-        ring_atom_ids = list(max(ssr, key=len))
+        return list(max(ssr, key=len))
 
-        return ring_atom_ids
-
-    def cycle_coords(self, path=None, conformer=-1):
+    def write_cycle_coords(self, path, conformer=-1):
         """
-        Find the coordinates of the macrocyclic atoms in the molecule.
+        Write the coords of the macrocyclic atoms to a file.
 
-        Coordinates of the atoms comprising the largest ring in the
-        macrocycle are found and the xyz coordinates file containing
-        only those atoms can be saved.
+        The coords are written in an ``.xyz`` format. For example::
 
-        Notes
-        -----
-        The approach identifies the Smallest Set of Symmetric Rings
-        and as a result one of multiple rings of the same size can be
-        chose arbitrarily, making the results not unique. This should
-        not be a problem in most applications.
+            atom_count
+
+            atom1_id x y z
+            atom2_id x y z
+            atom3_id x y z
 
         Parameters
         ----------
-        path : :class:`.str`
-            A path where the xyz file should be saved. If ``None`` then
-            no file is produced. InChKey is used as the filename.
+        path : :class:`str`
+            A path to the file.
+
+        conformer : :class:`int`, optional
+            The id of the conformer to be used.
 
         Returns
         -------
-        :class:`list` of :class:`list` of :class:`float`
-            Coordinates of the atoms in the largest ring in the format
-            ``[atom_index, x, y, z]``.
+        None : :class:`NoneType`
 
         """
 
-        ssr = rdkit.GetSymmSSSR(self.mol)
-        conf = self.mol.GetConformer(conformer)
-        macrocycle = (self.mol.GetAtomWithIdx(i)
-                      for i in max(ssr, key=len))
-        macro_coords = [[atom.GetIdx(),
-                         *conf.GetAtomPosition(atom.GetIdx())]
-                        for atom in macrocycle]
-
-        if path is not None:
-            name = rdkit.MolToInchiKey(self.mol)
-            xyz_file = f'{len(macro_coords)}\n\n'
-
-            for anum, *coords in macro_coords:
-                xyz_file += f'{anum} {coords[0]} {coords[1]} '
-                xyz_file += f'{coords[2]}\n'
-
-            if not os.path.exists(path):
-                os.makedirs(path)
-
-            with open(f'{path}/{name}.xyz', 'w') as f:
-                f.write(xyz_file)
-
-        return macro_coords
+        cycle_atoms = self.cycle_atoms(conformer)
+        atom_coords = (
+            (atom_id, *self.atom_coords(atom_id))
+            for atom_id in cycle_atoms
+        )
+        file_content = [f'{len(cycle_atoms)}\n']
+        file_content.extend(
+            f'{atom_id} {x} {y} {z}'
```

Comment:
That is not a legitimate xyz file. In order for the xyz file to be readable by an external program, it needs to have atomic_number rather than rdkit atom index.

Reply:
The first column should be element symbol, not atomic number. https://en.wikipedia.org/wiki/XYZ_file_format